src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,69 @@
+package duke.command;
+
+import java.util.Objects;
+
+import duke.task.Task;
+import duke.utils.CortanaException;
+import duke.utils.Storage;
+import duke.utils.TaskList;
+import duke.utils.Ui;
+
+/**
+ * The type Add command.
+ */
+public class AddCommand extends Command {
+    private final Task task;
+
+    /**
+     * Instantiates a new Add command.
+     *
+     * @param task the task to be added
+     */
+    public AddCommand(Task task) {
+        this.task = task;
+    }
+
+    /**
+     * Execute the add task operation.
+     *
+     * @param taskList the task list to operate on
+     * @param ui the ui to operate on
+     * @param storage the storage to operate on
+     */
+    public String execute(TaskList taskList, Ui ui, Storage storage) throws CortanaException {
+        if (taskList.getTaskSet().contains(task)) {
+            throw new CortanaException("Task already exists!");
+        } else {
+            taskList.getTaskSet().add(task);
+            taskList.getTaskList().add(task);
+            try {
+                storage.writeFile(taskList);
+            } catch (Exception e) {
+                throw new CortanaException(e.getMessage());
+            }
+            return ui.addedTask(taskList, task);
+        }
+    }
+
+    /**
+     * The program is not yet exited.
+     */
+    public boolean isExit() {
```

Comment:
~~Should it be named `isExit` if it always returns false?
Would a more suitable function name be `exitNow`?~~
Read below and found that current `isExit()` overrides the abstract method and there are multiple implementation of these

Reply:
Maybe you could annotate `@Override` so that it flags if you are not overriding the method in `Command` and readers would know there could be other implementation of `isExit()` 😃